modules/pulsar/src/main/java/org/testcontainers/containers/PulsarContainer.java

```diff
@@ -41,36 +51,65 @@ public PulsarContainer(String pulsarVersion) {
     public PulsarContainer(final DockerImageName dockerImageName) {
         super(dockerImageName);
         dockerImageName.assertCompatibleWith(DockerImageName.parse("apachepulsar/pulsar"));
-
         withExposedPorts(BROKER_PORT, BROKER_HTTP_PORT);
-        withCommand("/pulsar/bin/pulsar", "standalone", "--no-functions-worker", "-nss");
-        waitingFor(Wait.forHttp(METRICS_ENDPOINT).forStatusCode(200).forPort(BROKER_HTTP_PORT));
     }
 
     @Override
     protected void configure() {
         super.configure();
-
-        if (functionsWorkerEnabled) {
-            withCommand("/pulsar/bin/pulsar", "standalone");
-            waitingFor(
-                new WaitAllStrategy()
-                    .withStrategy(waitStrategy)
-                    .withStrategy(Wait.forLogMessage(".*Function worker service started.*", 1))
-            );
-        }
+        setupCommandAndEnv();
     }
 
     public PulsarContainer withFunctionsWorker() {
         functionsWorkerEnabled = true;
         return this;
     }
 
+    public PulsarContainer withTransactions() {
+        transactions = true;
+        return this;
+    }
+
+    public PulsarContainer withConfiguration(String name, String value) {
+        return withEnv("PULSAR_PREFIX_" + name, value);
+    }
+
+    public PulsarContainer withConfiguration(Map<String, String> configuration) {
+        configuration.forEach((name, value) -> withEnv("PULSAR_PREFIX_" + name, value));
+        return this;
+    }
```

Comment:
it is helpful because the right way to add a configuration property to Pulsar standalone is to set env with PULSAR_PREFIX_<configname>

otherwise the users have to remember to add the prefix

Reply:
Unfortunately I haven't found anything on the docs. 
This is the issue that implemented it: https://github.com/apache/pulsar/pull/3858

This is the script that does the work: https://github.com/apache/pulsar/blob/master/docker/pulsar/scripts/apply-config-from-env-with-prefix.py
and indeed I had to change the docker cmd in this pull: https://github.com/testcontainers/testcontainers-java/pull/5479/files#diff-87d7d34dfb8435aa9ab0532510e5109ae787d9b8126e398300293bf705b3cf4fR92

this is usually how you deploy real Pulsar cluster but the Pulsar standalone is meant for testing only so you need to explicity call the script

Reply:
upcoming pull that explain to PULSAR_PREFIX_ usage: https://github.com/apache/pulsar/pull/16085/files

Reply:
Thanks for the explanation. So I wasn't too incompetent to find the docs on `PULSAR_PREFIX`, they were simply not existing at this point in time 😅 

If I am honest, once the docs are in place in https://github.com/apache/pulsar/pull/16085, I think having the users configure it explicitly simply via `withEnv` seems Testcontainers idiomatic to me.

Reply:
What if I rename it `withPulsarEnv()`? 
I'm pretty sure people will forget to use the prefix.

Also, a nuance of Pulsar, if you set it without the prefix some properties work and other not 😞

Reply:
I can imagine that users might forget the prefix, but I can also imagine that users make typos if those values are just configured as Strings. Some other Testcontainers modules tend to use `enums` to mitigate this, but this approach comes with its own set of maintenance overhead.

In general, configuring Docker containers via ENV is idiomatic for Docker and for Testcontainers alike, and by removing the method, we don't unnecessarily increase the API size of the module. It is also pretty normal to have the technology-specific settings prefixed in some way (but it must admit the combination of `PULSAR_PREFIX` + camelCase config is a bit unusual to me).

Therefore, I think configuring those settings using the existing `withEnv()` method is the least surprising for Testcontainers users.